sirepo/template/flash.py

```diff
@@ -407,15 +406,20 @@ def _cell_size(f, refine_max):
     assert False, 'no blocks with appropriate refine level'
 
 
-def _extract_zip(data, run_dir):
-    import os
-    import sirepo.pkcli.flash
-    import stat
-    import zipfile
-    zipfile.ZipFile(run_dir.join(_SIM_DATA.proprietary_lib_file_basename(data))).extractall()
-    # extractall() doesn't maintain file permissions
-    # https://bugs.python.org/issue15795
-    os.chmod(run_dir.join(_SIM_DATA.EXE_NAME), stat.S_IXUSR)
+def _extract_rpm(data):
+    import subprocess
+
+    if _SIM_DATA.flash_exe_path(data):
+        return
+    #SECURITY: No user defined input in cmd so shell=True is ok
+    subprocess.check_output(
+        "rpm2cpio '{}' | cpio --extract --make-directories".format(
+            _SIM_DATA.lib_file_abspath(_SIM_DATA.FLASH_RPM),
```

Comment:
proprietary_code_rpm

Reply:
It is. The code your looking at is outdated.